src/memgraph.cpp

```diff
@@ -349,39 +349,54 @@ DEFINE_VALIDATED_string(log_level, "WARNING", log_level_help_string.c_str(), {
 });
 
 namespace {
-void ParseLogLevel() {
+spdlog::level::level_enum ParseLogLevel() {
   const auto log_level = StringToEnum<spdlog::level::level_enum>(FLAGS_log_level, log_level_mappings);
   MG_ASSERT(log_level, "Invalid log level");
-  spdlog::set_level(*log_level);
+  return *log_level;
 }
 
 // 5 weeks * 7 days
 constexpr auto log_retention_count = 35;
 
-void ConfigureLogging() {
-  std::vector<spdlog::sink_ptr> loggers;
+struct MemgraphLogger {
+  MemgraphLogger() : log_level(ParseLogLevel()) {
+    std::vector<spdlog::sink_ptr> sinks;
 
-  if (FLAGS_also_log_to_stderr) {
-    loggers.emplace_back(std::make_shared<spdlog::sinks::stderr_color_sink_mt>());
-  }
+    if (FLAGS_also_log_to_stderr) {
+      sinks.emplace_back(std::make_shared<spdlog::sinks::stderr_color_sink_mt>());
+    }
 
-  if (!FLAGS_log_file.empty()) {
-    // get local time
-    time_t current_time;
-    struct tm *local_time{nullptr};
+    if (!FLAGS_log_file.empty()) {
+      // get local time
+      time_t current_time{0};
+      struct tm *local_time{nullptr};
 
-    time(&current_time);
-    local_time = localtime(&current_time);
+      time(&current_time);
+      local_time = localtime(&current_time);
 
-    loggers.emplace_back(std::make_shared<spdlog::sinks::daily_file_sink_mt>(
-        FLAGS_log_file, local_time->tm_hour, local_time->tm_min, false, log_retention_count));
+      sinks.emplace_back(std::make_shared<spdlog::sinks::daily_file_sink_mt>(
+          FLAGS_log_file, local_time->tm_hour, local_time->tm_min, false, log_retention_count));
+    }
+    CreateLoggerFromSink(sinks);
```

Comment:
I would also move this to a separate function, because by creating the object, we can easily make another one obsolete.

In general I am not sure whether this object is a good abstraction. Multiple instances can be created, but all of them operate over a global resource (default logger). What I am thinking is tweaking this a bit:
1. Make the logger to be a singleton: this way it is explicitly expressed that only one object can exists.
2. Wrap the logic into a function: using function local static variables and `std::call_once` for initialization should be fine. As there is not need for destructor, this might be the easiest solution.

Reply:
Check it out now. What I wanted to avoid was fetching the logger every time we want to add a new sink because it takes a lock (same for the level). But it actually makes no sense as we shouldn't do that in any possible hot path + we do it only once now and it's during startup.

Reply:
I like this much better!